### PR TITLE
Build simd

### PR DESCRIPTION
**Pull request by huningxin**
Support SIMD option in JavaScript build and re-use intrin_cpp.hpp as base (fallback) of intrin_wasm.hpp.

### CODE REVIEW

**Comment by huningxin** [comment]
@Wenzhao-Xiang , PTAL. This is just a base. You can add your optimized implementation on top of that.

**Comment by huningxin** [comment]
Example usage
```
python ./platforms/js/build_js.py build_simd --build_wasm --build_test --build_perf --build_wasm_intrin_test --simd
```

File: modules/core/src/mathfuncs_core.simd.hpp [hunk 1]
```diff
@@ -32,6 +32,11 @@ using namespace cv;
 namespace {
 
 #ifdef __EMSCRIPTEN__
+static const float atan2_p1 = 0.9997878412794807f*(float)(180/CV_PI);
```
Comment (Wenzhao-Xiang) [inline modules/core/src/mathfuncs_core.simd.hpp:34]: Seems we can reuse the code below.

File: platforms/js/build_js.py [hunk 2]
```diff
@@ -150,6 +150,11 @@ def get_cmake_cmd(self):
         else:
             cmd.append("-DWITH_PTHREADS_PF=OFF")
 
+        if self.options.simd:
+            cmd.append("-DCV_ENABLE_INTRINSICS=ON")
+        else:
+            cmd.append("-DCV_ENABLE_INTRINSICS=OFF")
```
Comment (Wenzhao-Xiang) [inline platforms/js/build_js.py:156]: We may also need to pass the compile flag `-msimd128` to compiler.

**Review by Wenzhao-Xiang — CHANGES_REQUESTED** [review]

File: modules/core/src/mathfuncs_core.simd.hpp (hunk 1 above)
Comment (huningxin) [reply modules/core/src/mathfuncs_core.simd.hpp:34]: good catch, will do that.

File: platforms/js/build_js.py (hunk 2 above)
Comment (huningxin) [reply platforms/js/build_js.py:156]: yes, the base impl doesn't have any simd code, so I missed it. good catch. will add.

**Comment by huningxin** [comment]
@Wenzhao-Xiang , updated the PR, please take another look.

**Review by Wenzhao-Xiang — APPROVED** [review]

**Comment by Wenzhao-Xiang** [comment]
LGTM, thanks!

**Comment by huningxin** [comment]
Great. Merge it.

**Pull request merged**